### PR TITLE
✨ Add ignore patterns

### LINKED ISSUE #57: Feature request: ignore patterns for files or directories
Currently, bump-pydantic considers all files ending in `.py` when scanning and applying modifications. In a codebase where source files are located at the project root, and a virtual environment is present, bump-pydantic takes a very long time to go through everything in the virtual environment's `site-packages` in addition to the actual project source files.

Allowing the user to specify patterns to ignore when scanning, using a command-line argument, will solve this problem.

Further, some default patterns may be put in place, for example:

- Ignoring `.venv` as it is a common name for virtual environment directories, or
- Ignoring everything in `.gitignore` if it is present, if not falling back to the previous point

If the user wishes to suppress default ignore patterns, they might do so using `--ignore=""` or something similar.
These implementation details are up to the maintainers to decide.

If this feature is approved, I would like to work on it.

### PR DESCRIPTION
**Pull request by lemonyte**
Allows the user to specify glob patterns to ignore when scanning for source files.

- Closes #57 

Example usage:

```bash
# Ignore everything in the scripts directory.
bump-pydantic --ignore=scripts/** .
# Ignore only files in the tests directory, but don't ignore more deeply nested files.
bump-pydantic --ignore=tests/* .
# Ignore only files under directories named foo.
bump-pydantic --ignore=**/foo/* .
# Ignore all __init__.py files.
bump-pydantic --ignore=**/__init__.py .
```

Note: on POSIX systems it may be necessary to escape patterns in the command line.
```bash
bump-pydantic --ignore=scripts/\*\* .
```

## Implementation details and design decisions

- This PR adds a function `glob_to_re()` which translates a glob pattern into a regular expression. The function is necessary because:
  - the `glob` module does not provide a way to match a path against a pattern without traversing the file system, and
  - the `fnmatch.fnmatch()` and `pathlib.Path.match()` functions do not handle recursive wildcards (`**`).
 
- This feature requires disabling automatic argument expansion on Windows in Click. If not disabled, option strings like `.venv/**` will be transformed into `.venv\\` before bump-pydantic is able to apply its own interpretation of the string. The only way to pass this argument to Click is during the invocation of the app instance, therefore it was necessary to make the executable function a `functools.partial`.

- Extra: a list of default ignore patterns has been added, currently only containing `.venv/**`.

- Extra: if no files are left either because there are no source files in the given directory, or as a result of filtering with ignore patterns, the program now exits gracefully instead of having Mypy raise an error.

### CODE REVIEW

**Review by Kludex — CHANGES_REQUESTED** [review]
Can we add some unit tests for `glob_to_re`?

File: bump_pydantic/main.py [hunk 1]
```diff
@@ -26,9 +28,15 @@
     add_completion=False,
 )
 
+entrypoint = functools.partial(app, windows_expand_args=False)
```
[the remark this answers is not in the document]
Comment (lemonyte) [reply bump_pydantic/main.py:31]: Neither Typer nor Click seem to have an option to pass this argument during initialization. The only way to pass it is during invocation of the app through the `__call__()` or `main()` methods.

[the remark this answers is not in the document]
Comment (Kludex) [reply bump_pydantic/main.py:26]: Do you have a reference for this?

File: bump_pydantic/helpers.py
Comment (Kludex) [inline bump_pydantic/helpers.py:1]: Can we have a meaningful name instead of "helpers"?

**Review by Kludex — CHANGES_REQUESTED** [review]

File: bump_pydantic/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lemonyte) [reply bump_pydantic/main.py:26]: PR https://github.com/pallets/click/pull/1918 added the parameter, only to the `main` method.
The only mention of this parameter in Click's documentation is under the `main` method [here](https://click.palletsprojects.com/en/8.1.x/api/#click.BaseCommand.main).

File: bump_pydantic/helpers.py
Comment (lemonyte) [reply bump_pydantic/helpers.py:1]: Renamed to "glob_helpers".

**Review by Kludex — APPROVED** [review]
Thanks @lemonyte :)

**Pull request merged**